web_src/js/features/repo-issue-list.js

```diff
@@ -185,51 +185,24 @@ async function initIssuePinSort() {
   });
 }
 
-// function initArchivedLabelFilter() {
-//   const archivedLabelEl = document.querySelector('#archived-filter-checkbox');
-//   if (!archivedLabelEl) {
-//     return;
-//   }
-//
-//   const archivedLabels = $('[data-is-archived]');
-//   const archivedElToggle = () => {
-//     for (const label of archivedLabels) {
-//       toggleElem(label, archivedLabelEl.checked);
-//     }
-//   };
-//   archivedElToggle();
-//   archivedLabelEl.addEventListener('change', () => {
-//     archivedElToggle();
-//     const url = archivedLabelEl.getAttribute('data-url');
-//     window.location = (archivedLabelEl.checked) ? url : url.replace(/&archived=true/, '');
-//   });
-// }
 function initArchivedLabelFilter() {
   const archivedLabelEl = document.querySelector('#archived-filter-checkbox');
   if (!archivedLabelEl) {
     return;
   }
 
-  const archivedLabels = document.querySelectorAll('[data-is-archived]');
-  const toggleArchivedLabels = () => {
-    const isChecked = archivedLabelEl.checked;
+  const archivedLabels = $('[data-is-archived]');
+  const archivedElToggle = () => {
     for (const label of archivedLabels) {
-      toggleElem(label, isChecked);
+      toggleElem(label, archivedLabelEl.checked);
     }
   };
 
-  const updateURL = () => {
-    const url = archivedLabelEl.getAttribute('data-url');
-    window.location = archivedLabelEl.checked ? url : url.replace(/&archived=true/, '');
-  };
-
-  // Initial setup
-  toggleArchivedLabels();
-
-  // Event listeners
+  archivedElToggle();
   archivedLabelEl.addEventListener('change', () => {
-    toggleArchivedLabels();
-    updateURL();
+    archivedElToggle();
+    const url = archivedLabelEl.getAttribute('data-url');
+    window.location = (archivedLabelEl.checked) ? url : url.replace(/&archived=true/, '');
```

Comment:
This code will break as soon as the `archived` query param is at the beginning, btw.

Reply:
Thanks for pointing out this.